src/context.cc

```diff
@@ -120,6 +120,8 @@ Context::Context(const std::string &app_name, const std::string &app_version)
     resetLastTrackingReminderTime();
 
     pomodoro_break_entry_ = nullptr;
+
+    TogglClient::GetInstance().SetSyncStateMonitor(UI());
```

Comment:
I would follow the approach that we wouldn't introduce two singletons HTTPClient because we have two instances at the same time, it's not Singleton anymore.

On the other hand, we can add a boolean check to present SyncStateMonitor or not. The default value is true for  all requests, except a few requests.

Here is the list that doesn't show Sync from the master:
- `Context::downloadUpdate()`
- `Context::fetchMessage(const bool periodic)`
- `Context::pushObmAction()`
- `TimelineUploader::upload(TimelineBatch *batch)`

Reply:
@NghiaTranUIT Thanks for the fixes.
What I meant by 2 singletons was creating new classes, then it is a Singleton.
Here's my example: https://github.com/toggl-open-source/toggldesktop/commit/3c86a33d884df64ba8916283c8da3f6a972133f9
Besides, I've noticed we're using some instances of `HttpClient` which can cause some leaking. By introducing 2 classes we can use the one that doesn't use SyncMonitor instead of creating new instances of `HttpClient`.
Please have a look and consider this solution, I think it's nicer than introducing a boolean parameter to all methods.